diff --git a/dist/core.cjs.development.js b/dist/core.cjs.development.js
index 268502405d09f14023adf1d2be5a672efb353624..a410c91ab302f97d7aac45c1d44acc71bd8bde59 100644
--- a/dist/core.cjs.development.js
+++ b/dist/core.cjs.development.js
@@ -1825,10 +1825,6 @@ function useAutoScroller(_ref) {
     }
 
     for (const scrollContainer of sortedScrollableAncestors) {
-      if ((canScroll == null ? void 0 : canScroll(scrollContainer)) === false) {
-        continue;
-      }
-
       const index = scrollableAncestors.indexOf(scrollContainer);
       const scrollContainerRect = scrollableAncestorRects[index];
 
@@ -1841,6 +1837,10 @@ function useAutoScroller(_ref) {
         speed
       } = getScrollDirectionAndSpeed(scrollContainer, scrollContainerRect, rect, acceleration, threshold);
 
+      if ((canScroll == null ? void 0 : canScroll(scrollContainer, direction)) === false) {
+        continue;
+      }
+
       for (const axis of ['x', 'y']) {
         if (!scrollIntent[axis][direction[axis]]) {
           speed[axis] = 0;
diff --git a/dist/core.esm.js b/dist/core.esm.js
index 1577c253774e3c37eafa4f267f1944426deee405..979bba9f8d7fa54cab01e8e84643ffb0fb64c0d9 100644
--- a/dist/core.esm.js
+++ b/dist/core.esm.js
@@ -1824,10 +1824,6 @@ function useAutoScroller(_ref) {
     }
 
     for (const scrollContainer of sortedScrollableAncestors) {
-      if ((canScroll == null ? void 0 : canScroll(scrollContainer)) === false) {
-        continue;
-      }
-
       const index = scrollableAncestors.indexOf(scrollContainer);
       const scrollContainerRect = scrollableAncestorRects[index];
 
@@ -1840,6 +1836,10 @@ function useAutoScroller(_ref) {
         speed
       } = getScrollDirectionAndSpeed(scrollContainer, scrollContainerRect, rect, acceleration, threshold);
 
+      if ((canScroll == null ? void 0 : canScroll(scrollContainer, direction)) === false) {
+        continue;
+      }
+
       for (const axis of ['x', 'y']) {
         if (!scrollIntent[axis][direction[axis]]) {
           speed[axis] = 0;
diff --git a/dist/hooks/utilities/useAutoScroller.d.ts b/dist/hooks/utilities/useAutoScroller.d.ts
index 3159f04b27a9dad6b78d2c6d8d5df95c7e906ef2..01f66e3daaf7dc2f32495dddcd55d26c414f71ed 100644
--- a/dist/hooks/utilities/useAutoScroller.d.ts
+++ b/dist/hooks/utilities/useAutoScroller.d.ts
@@ -1,3 +1,4 @@
+import { Direction } from '../../types';
 import type { Coordinates, ClientRect } from '../../types';
 export declare type ScrollAncestorSortingFn = (ancestors: Element[]) => Element[];
 export declare enum AutoScrollActivator {
@@ -28,10 +29,14 @@ interface Arguments extends Options {
     scrollableAncestorRects: ClientRect[];
     delta: Coordinates;
 }
-export declare type CanScroll = (element: Element) => boolean;
+export declare type CanScroll = (element: Element, direction: ScrollDirection) => boolean;
 export declare enum TraversalOrder {
     TreeOrder = 0,
     ReversedTreeOrder = 1
 }
+interface ScrollDirection {
+    x: 0 | Direction;
+    y: 0 | Direction;
+}
 export declare function useAutoScroller({ acceleration, activator, canScroll, draggingRect, enabled, interval, order, pointerCoordinates, scrollableAncestors, scrollableAncestorRects, delta, threshold, }: Arguments): void;
 export {};
