diff --git a/node_modules/@dnd-kit/core/dist/core.esm.js b/node_modules/@dnd-kit/core/dist/core.esm.js
index 91671e9..2deb096 100644
--- a/node_modules/@dnd-kit/core/dist/core.esm.js
+++ b/node_modules/@dnd-kit/core/dist/core.esm.js
@@ -2348,7 +2348,7 @@ function useRects(elements, measure) {
     callback: measureRects
   });
 
-  if (elements.length > 0 && rects === defaultValue$2) {
+  if (elements.length !== rects.length) { // [CC]
     measureRects();
   }
 
