diff --git a/node_modules/react-data-grid/lib/bundle.cjs b/node_modules/react-data-grid/lib/bundle.cjs
index 77f0005..7417a25 100644
--- a/node_modules/react-data-grid/lib/bundle.cjs
+++ b/node_modules/react-data-grid/lib/bundle.cjs
@@ -2859,6 +2859,7 @@ exports.headerRenderer = headerRenderer;
 exports.sortIcon = sortIcon;
 exports.sortPriority = sortPriority;
 exports.textEditor = textEditor;
+exports.textEditorClassname = textEditorClassname; // [CC]
 exports.toggleGroupFormatter = toggleGroupFormatter;
 exports.useRowSelection = useRowSelection;
 exports.valueFormatter = valueFormatter;
diff --git a/node_modules/react-data-grid/lib/bundle.js b/node_modules/react-data-grid/lib/bundle.js
index b9b5e7d..3f69655 100644
--- a/node_modules/react-data-grid/lib/bundle.js
+++ b/node_modules/react-data-grid/lib/bundle.js
@@ -2843,5 +2843,6 @@ function textEditor({
   });
 }
 
-export { DataGridDefaultRenderersProvider, RowComponent$1 as Row, SELECT_COLUMN_KEY, SelectCellFormatter, SelectColumn, ToggleGroup, checkboxFormatter, DataGrid$1 as default, headerRenderer, sortIcon, sortPriority, textEditor, toggleGroupFormatter, useRowSelection, valueFormatter };
+// [CC] (patch-package) export textEditorClassname
+export { DataGridDefaultRenderersProvider, RowComponent$1 as Row, SELECT_COLUMN_KEY, SelectCellFormatter, SelectColumn, ToggleGroup, checkboxFormatter, DataGrid$1 as default, headerRenderer, sortIcon, sortPriority, textEditor, textEditorClassname, toggleGroupFormatter, useRowSelection, valueFormatter };
 //# sourceMappingURL=bundle.js.map
diff --git a/node_modules/react-data-grid/lib/index.d.ts b/node_modules/react-data-grid/lib/index.d.ts
index 4e19c35..46c160e 100644
--- a/node_modules/react-data-grid/lib/index.d.ts
+++ b/node_modules/react-data-grid/lib/index.d.ts
@@ -398,6 +398,8 @@ export declare interface SummaryFormatterProps<TSummaryRow, TRow = unknown> {
     tabIndex: number;
 }
 
+export declare const textEditorClassname: string; // [CC]
+
 export declare function textEditor<TRow, TSummaryRow>({ row, column, onRowChange, onClose }: EditorProps<TRow, TSummaryRow>): JSX_2.Element;
 
 export declare function ToggleGroup<R, SR>({ groupKey, isExpanded, tabIndex, toggleGroup }: GroupFormatterProps<R, SR>): JSX_2.Element;
