diff --git a/node_modules/react-data-grid/lib/bundle.js b/node_modules/react-data-grid/lib/bundle.js
index dfb48e8..46ab98a 100644
--- a/node_modules/react-data-grid/lib/bundle.js
+++ b/node_modules/react-data-grid/lib/bundle.js
@@ -3011,5 +3011,6 @@ function TextEditor({
   });
 }
 
-export { CheckboxFormatter, DataGridDefaultComponentsProvider, HeaderRenderer, Row$1 as Row, SELECT_COLUMN_KEY, SelectCellFormatter, SelectColumn, TextEditor, ToggleGroupFormatter, ValueFormatter, DataGrid$1 as default, useRowSelection };
+// [CC] (patch-package) export internal class of TextEditor
+export { CheckboxFormatter, DataGridDefaultComponentsProvider, HeaderRenderer, Row$1 as Row, SELECT_COLUMN_KEY, SelectCellFormatter, SelectColumn, textEditor as textEditorClassCC, TextEditor, ToggleGroupFormatter, ValueFormatter, DataGrid$1 as default, useRowSelection };
 //# sourceMappingURL=bundle.js.map
diff --git a/node_modules/react-data-grid/lib/index.d.ts b/node_modules/react-data-grid/lib/index.d.ts
index 1aa42fd..a0f3ad7 100644
--- a/node_modules/react-data-grid/lib/index.d.ts
+++ b/node_modules/react-data-grid/lib/index.d.ts
@@ -346,6 +346,9 @@ export declare interface SummaryFormatterProps<TSummaryRow, TRow = unknown> {
     isCellSelected: boolean;
 }
 
+// [CC] (patch-package) export internal class of TextEditor
+export declare const textEditorClassCC: string;
+
 export declare function TextEditor<TRow, TSummaryRow>({ row, column, onRowChange, onClose }: EditorProps<TRow, TSummaryRow>): JSX.Element;
 
 export declare function ToggleGroupFormatter<R, SR>({ groupKey, isExpanded, isCellSelected, toggleGroup }: GroupFormatterProps<R, SR>): JSX.Element;
