diff --git a/node_modules/react-data-grid/lib/bundle.cjs b/node_modules/react-data-grid/lib/bundle.cjs
index b3e8235..bd4ac8e 100644
--- a/node_modules/react-data-grid/lib/bundle.cjs
+++ b/node_modules/react-data-grid/lib/bundle.cjs
@@ -2845,5 +2845,6 @@ exports.renderSortPriority = renderSortPriority;
 exports.renderToggleGroup = renderToggleGroup;
 exports.renderValue = renderValue;
 exports.textEditor = textEditor;
+exports.textEditorClassname = textEditorClassname; // [CC]
 exports.useRowSelection = useRowSelection;
 //# sourceMappingURL=bundle.cjs.map
diff --git a/node_modules/react-data-grid/lib/bundle.js b/node_modules/react-data-grid/lib/bundle.js
index 6460beb..97d9a5e 100644
--- a/node_modules/react-data-grid/lib/bundle.js
+++ b/node_modules/react-data-grid/lib/bundle.js
@@ -2827,5 +2827,6 @@ function textEditor({
   });
 }
 
-export { DataGridDefaultRenderersProvider, RowComponent$1 as Row, SELECT_COLUMN_KEY, SelectCellFormatter, SelectColumn, ToggleGroup, DataGrid$1 as default, renderCheckbox, renderHeaderCell, renderSortIcon, renderSortPriority, renderToggleGroup, renderValue, textEditor, useRowSelection };
+// [CC] (patch-package) export textEditorClassname
+export { DataGridDefaultRenderersProvider, RowComponent$1 as Row, SELECT_COLUMN_KEY, SelectCellFormatter, SelectColumn, ToggleGroup, DataGrid$1 as default, renderCheckbox, renderHeaderCell, renderSortIcon, renderSortPriority, renderToggleGroup, renderValue, textEditor, textEditorClassname, useRowSelection };
 //# sourceMappingURL=bundle.js.map
diff --git a/node_modules/react-data-grid/lib/index.d.ts b/node_modules/react-data-grid/lib/index.d.ts
index cec4c95..0b16535 100644
--- a/node_modules/react-data-grid/lib/index.d.ts
+++ b/node_modules/react-data-grid/lib/index.d.ts
@@ -407,6 +407,8 @@ export declare interface SortColumn {
 
 export declare type SortDirection = 'ASC' | 'DESC';
 
+export declare const textEditorClassname: string; // [CC]
+
 export declare function textEditor<TRow, TSummaryRow>({ row, column, onRowChange, onClose }: RenderEditCellProps<TRow, TSummaryRow>): JSX_2.Element;
 
 export declare function ToggleGroup<R, SR>({ groupKey, isExpanded, tabIndex, toggleGroup }: RenderGroupCellProps<R, SR>): JSX_2.Element;
